lib/util.js

```diff
@@ -51,13 +55,17 @@ function _testSetHapi18(isHapi18Flag) {
 /**
  * Detects the installed Hapi version and use the appropriate plugin
  *
- * @param {*} registers: { hapi16, hapi17 } contains registers for hapi16, hapi17/18
+ * @param {Object} registers: { hapi16, hapi17 } contains registers for hapi16, hapi17/18
+ * @param {*|undefined} registers.fastify: optional field for fastify plugin
  * @param {*} pkg : package for a hapi plugin
  * @returns {*}: Hapi plugin appropriate for installed version
  *
  */
 function universalHapiPlugin(registers, pkg) {
-  if (hapiMajor() >= 17) {
+  if (registers.fastify && isFastify()) {
```

Comment:
if fastify, we want to error if they don't specify registers.fastify. 
with current code, we return the hapi16 function, which would probably result in a weird error.

Reply:
Makes sense